pkg/codegen/schema.go

```diff
@@ -401,8 +401,21 @@ func GenerateGoSchema(sref *openapi3.SchemaRef, path []string) (Schema, error) {
 		for i, enumValue := range schema.Enum {
 			enumValues[i] = fmt.Sprintf("%v", enumValue)
 		}
+		if 0 < len(schema.ExtensionProps.Extensions) {
+			//fmt.Fprintf(os.Stderr, "%#v\n\n", schema)
+		}
+
+		enumNames := enumValues
+		for _, key := range []string{extEnumVarNames, extEnumNames} {
+			if _, ok := schema.ExtensionProps.Extensions[key]; ok {
+				if extEnumNames, err := extParseEnumVarNames(schema.ExtensionProps.Extensions[key]); err == nil {
+					enumNames = extEnumNames
```

Comment:
Right now, your extension completely replaces the enum names that are autogenerated. What happens when the provided names array has a different length than possible enum values? I think we'll end up creating invalid enums. Can you, at a minimum, error out on cardinality mismatch?